src/editors/containers/ProblemEditor/components/SelectTypeModal/content/AdvanceTypeSelect.tsx

```diff
@@ -54,25 +55,29 @@ const AdvanceTypeSelect: React.FC<Props> = ({
           className="px-4"
         >
           {Object.entries(AdvanceProblems).map(([type, data]) => {
-            if (data.status !== '') {
+            if (data.status !== AdvanceProblemsStatus.empty) {
               return (
                 <ActionRow className="border-primary-100 border-bottom m-0 py-3 w-100" key={type}>
                   <Form.Radio id={type} value={type}>
-                    {intl.formatMessage(messages.advanceProblemTypeLabel, { problemType: data.title })}
+                    {intl.formatMessage(messages.advanceProblemTypeLabel, {
+                      problemType: intl.formatMessage(data.titleMessage),
```

Comment:
Could you elaborate on the `title` vs `titleMessage` stuff? I see down in `problem.ts` you've added `titleMessage` and set `title` to use the `defaultMessage`.

Reply:
The idea of `titleMessage` was to prevent having to create further code refactor on the references of the `problem.ts` file.
But still kind of using the existing design of Object Oriented idea of the `ProblemTypes` and `AdvanceProblems`.
And extracting the i18n to a proper `defineMessages` so the `formatjs` can detect them.
So, when you want to present them to the user, I just replace the `title` with `titleMessage` on a `intl.formatMessage(...)`.